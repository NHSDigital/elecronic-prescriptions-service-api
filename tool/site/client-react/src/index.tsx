import * as React from "react"
import {PageContainer} from "./components/pageContainer"
import PrescriptionSummaryPage from "./pages/prescriptionSummaryPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import SendPage from "./pages/sendPage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/"})

async function startApplication(baseUrl: string): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl}}>
      <PageContainer>
        <BrowserRouter>
          <Switch>
            <Route path={`${baseUrl}prescribe/edit`}>
              <PrescriptionSummaryPage prescriptionId={urlParams.get("prescription_id")}/>
            </Route>
            <Route path={`${baseUrl}prescribe/send`}>
              <SendPage token={urlParams.get("token")}/>
            </Route>
            <Route path={`${baseUrl}dispense/dispense`}>
              <DispensePage prescriptionId={urlParams.get("prescription_id")}/>
            </Route>
            <Route path={`${baseUrl}dispense/claim`}>
              <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
            </Route>
            <Route path={`${baseUrl}search`}>
              <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")}/>
            </Route>
          </Switch>
        </BrowserRouter>
      </PageContainer>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
