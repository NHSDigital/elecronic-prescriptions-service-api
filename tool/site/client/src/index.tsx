import * as React from "react"
import {Page} from "./components/page"
import SendPreSignPage from "./pages/sendPreSignPage"
import * as ReactDOM from "react-dom"
import {BrowserRouter, Route, Switch} from "react-router-dom"
import ClaimPage from "./pages/claimPage"
import DispensePage from "./pages/dispensePage"
import {CookiesProvider} from "react-cookie"
import SendPostSignPage from "./pages/sendPostSignPage"
import HomePage from "./pages/homePage"
import PrescriptionSearchPage from "./pages/prescriptionSearchPage"
import ReleasePage from "./pages/releasePage"
import CancelPage from "./pages/cancelPage"
import MyPrescriptionsPage from "./pages/myPrescriptionsPage"
import LoginPage from "./pages/loginPage"
import ValidatePage from "./pages/validatePage"
import ReturnPage from "./pages/returnPage"
import WithdrawPage from "./pages/withdrawPage"
import LoadPage from "./pages/loadPage"
import {Environment} from "./services/environment"
import LogoutPage from "./pages/logoutPage"
import ConfigPage from "./pages/configPage"
import VerifyPage from "./pages/verifyPage"
import ComparePage from "./pages/comparePage"
import {PageFullWidth} from "./components/pageFullWidth"
import {PageContainer} from "./components/pageContainer"

const customWindow = window as Record<string, any>

export interface AppContextValue {
  baseUrl: string
  environment: Environment
}

export const AppContext = React.createContext<AppContextValue>({baseUrl: "/", environment: "int"})

async function startApplication(baseUrl: string, environment: Environment): Promise<void> {
  const urlParams = new URLSearchParams(window.location.search)
  const content = (
    <AppContext.Provider value={{baseUrl, environment}}>
      <CookiesProvider>
        <Page>
          <BrowserRouter>
            <Switch>
              <Route path={`${baseUrl}/`}>
                <PageContainer>
                  <HomePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}config`}>
                <PageContainer>
                  <ConfigPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}my-prescriptions`}>
                <PageContainer>
                  <MyPrescriptionsPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}login`}>
                <PageContainer>
                  <LoginPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}logout`}>
                <PageContainer>
                  <LogoutPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/load`}>
                <PageContainer>
                  <LoadPage/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/edit`}>
                <PageContainer>
                  <SendPreSignPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/send`}>
                <PageContainer>
                  <SendPostSignPage token={urlParams.get("token")} state={urlParams.get("state")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}prescribe/cancel`}>
                <PageContainer>
                  <CancelPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/release`}>
                <PageContainer>
                  <ReleasePage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/verify`}>
                <PageContainer>
                  <VerifyPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/return`}>
                <PageContainer>
                  <ReturnPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/withdraw`}>
                <PageContainer>
                  <WithdrawPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/dispense`}>
                <PageContainer>
                  <DispensePage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}dispense/claim`}>
                <PageContainer>
                  <ClaimPage prescriptionId={urlParams.get("prescription_id")}/>
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}search`}>
                <PageContainer>
                  <PrescriptionSearchPage prescriptionId={urlParams.get("prescription_id")} />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}validate`}>
                <PageContainer>
                  <ValidatePage />
                </PageContainer>
              </Route>
              <Route path={`${baseUrl}compare-prescriptions`}>
                <PageFullWidth>
                  <ComparePage />
                </PageFullWidth>
              </Route>
            </Switch>
          </BrowserRouter>
        </Page>
      </CookiesProvider>
    </AppContext.Provider>
  )
  ReactDOM.render(content, document.getElementById("root"))
}

customWindow.startApplication = startApplication
