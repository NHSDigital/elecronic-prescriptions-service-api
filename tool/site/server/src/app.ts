import Hapi from "@hapi/hapi"
import routes from "./routes"
import HapiPino from "hapi-pino"
import Vision from "@hapi/vision"
import * as inert from "@hapi/inert"
import Yar from "@hapi/yar"
import Cookie from "@hapi/cookie"
import {isDev, isLocal} from "./services/environment"
import axios from "axios"
import {setSessionValue} from "./services/session"

const init = async () => {
  axios.defaults.validateStatus = () => true

  const server = Hapi.server({
    port: 9000,
    host: "0.0.0.0",
    routes: {
      cors: true // Won't run as Apigee hosted target without this
    }
  })

  const baseUrl = process.env.BASE_PATH
    ? `/${process.env.BASE_PATH}/`
    : "/"

  await server.register(Cookie)
  server.auth.strategy("session", "cookie", {
    cookie: {
      name: "auth",
      password: process.env.SESSION_TOKEN_ENCRYPTION_KEY,
      isSecure: true
    },
    redirectTo: (request: Hapi.Request) => {
      if (isDev()) {
        setSessionValue(
          "use_signing_mock",
          request.query["use_signing_mock"],
          request
        )
      }
      return `${baseUrl}login`
    }
  })
  server.auth.default("session")

  server.route(routes)

  await server.register({
    plugin: Yar,
    options: {
      storeBlank: false,
      // Use "0" maxCookieSize to force all session data to be written to cache
      maxCookieSize: 0,
      cache: {
        expiresIn: 24 * 60 * 60 * 1000
      },
      cookieOptions: {
        password: process.env.SESSION_TOKEN_ENCRYPTION_KEY ?? "",
        isSecure: true,
        isSameSite: "None"
      }
    }
  })

  await server.register({
    plugin: HapiPino,
    options: {
      // Pretty print in local environment only to avoid spamming logs
      prettyPrint: isLocal(),
      // Redact Authorization headers, see https://getpino.io/#/docs/redaction
      redact: ["req.headers.authorization"]
    }
  })

  await server.register(inert)

  server.route({
    method: "GET",
    path: `/static/{param*}`,
    options: {
      auth: false
    },
    handler: {
      directory: {
        path: "static"
      }
    }
  })

  await server.register(Vision)

  server.views({
    engines: {
      html: require("handlebars")
    },
    relativeTo: __dirname,
    path: "templates"
  })

  server.route(addHomeViewRoute())
  server.route(addViewRoute("login", true))
  server.route(addViewRoute("my-prescriptions"))
  server.route(addViewRoute("validate"))
  server.route(addViewRoute("search"))
  server.route(addViewRoute("prescribe/load"))
  server.route(addViewRoute("prescribe/edit"))
  server.route(addViewRoute("prescribe/send"))
  server.route(addViewRoute("prescribe/cancel"))
  server.route(addViewRoute("dispense/release"))
  server.route(addViewRoute("dispense/return"))
  server.route(addViewRoute("dispense/dispense"))
  server.route(addViewRoute("dispense/withdraw"))
  server.route(addViewRoute("dispense/claim"))

  function addHomeViewRoute() : Hapi.ServerRoute {

    const baseUrl = process.env.BASE_PATH
      ? `/${process.env.BASE_PATH}/`
      : "/"

    return {
      method: "GET",
      path: `/`,
      handler: {
        view: {
          template: "index",
          context: {
            baseUrl,
            environment: process.env.ENVIRONMENT
          }
        }
      }
    }
  }

  function addViewRoute(path: string, skipAuth?: boolean): Hapi.ServerRoute {
    const baseUrl = process.env.BASE_PATH
      ? `/${process.env.BASE_PATH}/`
      : "/"

    const viewRoute = {
      method: "GET",
      path: `/${path}`,
      handler: {
        view: {
          template: "index",
          context: {
            baseUrl,
            environment: process.env.ENVIRONMENT
          }
        }
      }
    }

    if (skipAuth) {
      return {
        ...viewRoute,
        options: {
          auth: false
        }
      }
    }

    return viewRoute
  }

  await server.start()
  server.log("info", `Server running on ${server.info.uri}`)
}

process.on("unhandledRejection", err => {
  console.log(err)
  process.exit(1)
})

init()
