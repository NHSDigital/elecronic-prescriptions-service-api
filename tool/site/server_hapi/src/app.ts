import Hapi from "@hapi/hapi"
import routes from "./routes"
import HapiPino from "hapi-pino"

const init = async () => {
  const server = Hapi.server({
    port: 9002,
    host: "0.0.0.0",
    routes: {
      cors: true, // Won't run as Apigee hosted target without this
      payload: {
        parse: false
      }
    }
  })

  server.route(routes)

  await server.register({
    plugin: HapiPino,
    options: {
      // Dont pretty print to avoid spamming logs
      prettyPrint: false,
      // Redact Authorization headers, see https://getpino.io/#/docs/redaction
      redact: ["req.headers.authorization"]
    }
  })

  await server.start()
  server.log("info", `Server running on ${server.info.uri}`)
}

process.on("unhandledRejection", (err) => {
  console.log(err)
  process.exit(1)
})

init()
