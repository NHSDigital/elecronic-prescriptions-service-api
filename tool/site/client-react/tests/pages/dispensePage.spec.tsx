import {waitFor} from "@testing-library/react"
import {screen} from "@testing-library/dom"
import pretty from "pretty"
import * as React from "react"
import moxios from "moxios"
import userEvent from "@testing-library/user-event"
import {readMessage} from "../messages/messages"
import {AppContextValue} from "../../src"
import {renderWithContext} from "../renderWithContext"
import DispensePage from "../../src/pages/dispensePage"

const baseUrl = "baseUrl/"
const prescriptionId = "7A9089-A83008-56A03J"
const context: AppContextValue = {baseUrl}

const prescriptionOrderUrl = `${baseUrl}prescription/${prescriptionId}`
const dispenseNotificationUrl = `${baseUrl}dispenseNotifications/${prescriptionId}`
const dispenseUrl = `${baseUrl}dispense/dispense`

const prescriptionOrder = readMessage("prescriptionOrder.json")
const dispenseNotification = readMessage("dispenseNotification.json")

beforeEach(() => moxios.install())

afterEach(() => moxios.uninstall())

test("Displays loading text while prescription data is being requested", async () => {
  const {container} = renderWithContext(<DispensePage prescriptionId={prescriptionId}/>, context)
  await waitFor(() => screen.getByText("Retrieving prescription details."))

  expect(screen.getByText("Loading...")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays dispense form if prescription details are retrieved successfully (no previous dispense notification)", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: []
  })

  const container = await renderPage()

  expect(screen.getByText("Dispense")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays dispense form if prescription details are retrieved successfully (previous dispense notification)", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: [dispenseNotification]
  })

  const container = await renderPage()

  expect(screen.getByText("Dispense")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays an error if prescription-order not found", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: null
  })

  const {container} = renderWithContext(<DispensePage prescriptionId={prescriptionId}/>, context)
  await waitFor(() => screen.getByText("Error"))

  expect(screen.getByText("Prescription order not found. Is the ID correct?")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays an error on invalid response", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 500,
    response: {}
  })

  const {container} = renderWithContext(<DispensePage prescriptionId={prescriptionId}/>, context)
  await waitFor(() => screen.getByText("Error"))

  expect(screen.getByText("Request failed with status code 500")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays loading text while dispense notification is being submitted", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: [dispenseNotification]
  })

  const container = await renderPage()
  userEvent.click(screen.getByText("Dispense"))
  await waitFor(() => screen.getByText("Loading..."))

  expect(screen.getByText("Sending dispense notification.")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

test("Displays dispense result", async () => {
  moxios.stubRequest(prescriptionOrderUrl, {
    status: 200,
    response: prescriptionOrder
  })
  moxios.stubRequest(dispenseNotificationUrl, {
    status: 200,
    response: [dispenseNotification]
  })
  moxios.stubRequest(dispenseUrl, {
    status: 200,
    response: {
      success: true,
      request: "JSON Request",
      request_xml: "XML Request",
      response: "JSON Response",
      response_xml: "XML Response"
    }
  })

  const container = await renderPage()
  userEvent.click(screen.getByText("Dispense"))
  await waitFor(() => screen.getByText(/Dispense Result/))

  expect(screen.getByText(JSON.stringify("JSON Request"))).toBeTruthy()
  expect(screen.getByText("XML Request")).toBeTruthy()
  expect(screen.getByText(JSON.stringify("JSON Response"))).toBeTruthy()
  expect(screen.getByText("XML Response")).toBeTruthy()
  expect(pretty(container.innerHTML)).toMatchSnapshot()
})

async function renderPage() {
  const {container} = renderWithContext(<DispensePage prescriptionId={prescriptionId}/>, context)
  await waitFor(() => screen.getByText("Dispense Medication"))
  return container
}
