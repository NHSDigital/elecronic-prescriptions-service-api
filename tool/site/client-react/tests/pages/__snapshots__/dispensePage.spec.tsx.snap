// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Displays an error if prescription-order not found 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Error</label></h1><span class=\\"nhsuk-error-message\\" role=\\"alert\\"><span class=\\"nhsuk-u-visually-hidden\\">Error: </span>Prescription order not found. Is the ID correct?</span>
<div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><a class=\\"nhsuk-button nhsuk-button--secondary\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"baseUrl/\\">Back</a></div>"
`;

exports[`Displays an error on invalid response 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Error</label></h1><span class=\\"nhsuk-error-message\\" role=\\"alert\\"><span class=\\"nhsuk-u-visually-hidden\\">Error: </span>Request failed with status code 500</span>
<div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><a class=\\"nhsuk-button nhsuk-button--secondary\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"baseUrl/\\">Back</a></div>"
`;

exports[`Displays dispense form if prescription details are retrieved successfully (no previous dispense notification) 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Dispense Medication</label></h1>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Methotrexate 10mg/0.2ml solution for injection pre-filled syringes</legend>
      <dl class=\\"nhsuk-summary-list nhsuk-summary-list--no-border\\">
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Quantity Requested</dt>
          <dd class=\\"nhsuk-summary-list__value\\">63 tablet</dd>
        </div>
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Prior Status</dt>
          <dd class=\\"nhsuk-summary-list__value\\">Item to be dispensed</dd>
        </div>
      </dl>
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label\\" id=\\"lineItems.0.statusCode--label\\" for=\\"lineItems.0.statusCode\\">Status</label><select class=\\"nhsuk-select\\" aria-labelledby=\\"lineItems.0.statusCode--label\\" name=\\"lineItems.0.statusCode\\" id=\\"lineItems.0.statusCode\\">
          <option value=\\"0001\\">Item fully dispensed</option>
          <option value=\\"0002\\">Item not dispensed</option>
          <option value=\\"0003\\">Item dispensed - partial</option>
          <option value=\\"0004\\">Item not dispensed owing</option>
          <option value=\\"0005\\">Item cancelled</option>
          <option value=\\"0006\\">Expired</option>
          <option value=\\"0007\\">Item to be dispensed</option>
          <option value=\\"0008\\">Item with dispenser</option>
        </select></div>
    </fieldset>
  </div>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Prescription Status</legend>
      <dl class=\\"nhsuk-summary-list nhsuk-summary-list--no-border\\">
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Prior Status</dt>
          <dd class=\\"nhsuk-summary-list__value\\">To be Dispensed</dd>
        </div>
      </dl>
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label\\" id=\\"prescription.statusCode--label\\" for=\\"prescription.statusCode\\">Status</label><select class=\\"nhsuk-select\\" aria-labelledby=\\"prescription.statusCode--label\\" name=\\"prescription.statusCode\\" id=\\"prescription.statusCode\\">
          <option value=\\"0001\\">To be Dispensed</option>
          <option value=\\"0002\\">With Dispenser</option>
          <option value=\\"0003\\">With Dispenser - Active</option>
          <option value=\\"0004\\">Expired</option>
          <option value=\\"0005\\">Cancelled</option>
          <option value=\\"0006\\">Dispensed</option>
          <option value=\\"0007\\">Not Dispensed</option>
        </select></div>
    </fieldset>
  </div>
  <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Dispense</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"reset\\">Reset</button></div>
</form>"
`;

exports[`Displays dispense form if prescription details are retrieved successfully (previous dispense notification) 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Dispense Medication</label></h1>
<form>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Methotrexate 10mg/0.2ml solution for injection pre-filled syringes</legend>
      <dl class=\\"nhsuk-summary-list nhsuk-summary-list--no-border\\">
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Quantity Requested</dt>
          <dd class=\\"nhsuk-summary-list__value\\">63 tablet</dd>
        </div>
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Prior Status</dt>
          <dd class=\\"nhsuk-summary-list__value\\">Item fully dispensed</dd>
        </div>
      </dl>
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label\\" id=\\"lineItems.0.statusCode--label\\" for=\\"lineItems.0.statusCode\\">Status</label><select class=\\"nhsuk-select\\" aria-labelledby=\\"lineItems.0.statusCode--label\\" name=\\"lineItems.0.statusCode\\" id=\\"lineItems.0.statusCode\\">
          <option value=\\"0001\\">Item fully dispensed</option>
          <option value=\\"0002\\">Item not dispensed</option>
          <option value=\\"0003\\">Item dispensed - partial</option>
          <option value=\\"0004\\">Item not dispensed owing</option>
          <option value=\\"0005\\">Item cancelled</option>
          <option value=\\"0006\\">Expired</option>
          <option value=\\"0007\\">Item to be dispensed</option>
          <option value=\\"0008\\">Item with dispenser</option>
        </select></div>
    </fieldset>
  </div>
  <div class=\\"nhsuk-form-group\\">
    <fieldset class=\\"nhsuk-fieldset\\">
      <legend class=\\"nhsuk-fieldset__legend nhsuk-fieldset__legend--m\\">Prescription Status</legend>
      <dl class=\\"nhsuk-summary-list nhsuk-summary-list--no-border\\">
        <div class=\\"nhsuk-summary-list__row\\">
          <dt class=\\"nhsuk-summary-list__key\\">Prior Status</dt>
          <dd class=\\"nhsuk-summary-list__value\\">Dispensed</dd>
        </div>
      </dl>
      <div class=\\"nhsuk-form-group\\"><label class=\\"nhsuk-label\\" id=\\"prescription.statusCode--label\\" for=\\"prescription.statusCode\\">Status</label><select class=\\"nhsuk-select\\" aria-labelledby=\\"prescription.statusCode--label\\" name=\\"prescription.statusCode\\" id=\\"prescription.statusCode\\">
          <option value=\\"0001\\">To be Dispensed</option>
          <option value=\\"0002\\">With Dispenser</option>
          <option value=\\"0003\\">With Dispenser - Active</option>
          <option value=\\"0004\\">Expired</option>
          <option value=\\"0005\\">Cancelled</option>
          <option value=\\"0006\\">Dispensed</option>
          <option value=\\"0007\\">Not Dispensed</option>
        </select></div>
    </fieldset>
  </div>
  <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\">Dispense</button><button class=\\"nhsuk-button nhsuk-button--secondary\\" aria-disabled=\\"false\\" type=\\"reset\\">Reset</button></div>
</form>"
`;

exports[`Displays dispense result 1`] = `
"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Dispense Result <svg class=\\"nhsuk-icon nhsuk-icon__tick\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path stroke-width=\\"4\\" stroke-linecap=\\"round\\" stroke=\\"#007f3b\\" fill=\\"none\\" d=\\"M18.4 7.8l-8.5 8.4L5.6 12\\"></path>
    </svg></label></h1>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/dispense?prescription_id=7A9089-A83008-56A03J\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Dispense prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/dispense/claim?prescription_id=7A9089-A83008-56A03J\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">Claim for prescription</span></a></div>
<div class=\\"nhsuk-action-link\\"><a class=\\"nhsuk-action-link__link\\" href=\\"baseUrl/search?prescription_id=7A9089-A83008-56A03J\\"><svg class=\\"nhsuk-icon nhsuk-icon__arrow-right-circle\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" height=\\"32\\" width=\\"32\\">
      <path d=\\"M0 0h24v24H0z\\" fill=\\"none\\"></path>
      <path d=\\"M12 2a10 10 0 0 0-9.95 9h11.64L9.74 7.05a1 1 0 0 1 1.41-1.41l5.66 5.65a1 1 0 0 1 0 1.42l-5.66 5.65a1 1 0 0 1-1.41 0 1 1 0 0 1 0-1.41L13.69 13H2.05A10 10 0 1 0 12 2z\\"></path>
    </svg><span class=\\"nhsuk-action-link__text\\">View prescription</span></a></div>
<div class=\\"nhsuk-expander-group\\">
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Request%22\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">\\"JSON Request\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Request (HL7 V3)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Request\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">XML Request</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (FHIR)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:application/json;charset=utf-8,%22JSON%20Response%22\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">\\"JSON Response\\"</pre>
    </div>
  </details>
  <details class=\\"nhsuk-details nhsuk-expander\\">
    <summary class=\\"nhsuk-details__summary\\"><span class=\\"nhsuk-details__summary-text\\">Response (HL7 V3)</span></summary>
    <div class=\\"nhsuk-details__text\\">
      <div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><button class=\\"nhsuk-button\\" aria-disabled=\\"false\\" type=\\"submit\\" style=\\"margin-bottom: 0px;\\">Copy</button><a class=\\"nhsuk-button\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"data:text/xml;charset=utf-8,XML%20Response\\" style=\\"margin-bottom: 0px;\\" download=\\"message\\">Download</a></div><pre style=\\"white-space: break-spaces; overflow-wrap: anywhere;\\">XML Response</pre>
    </div>
  </details>
</div>
<div style=\\"display: flex; flex-direction: row; gap: 0.5em;\\"><a class=\\"nhsuk-button nhsuk-button--secondary\\" role=\\"button\\" aria-disabled=\\"false\\" draggable=\\"false\\" href=\\"baseUrl/\\" type=\\"button\\">Back</a></div>"
`;

exports[`Displays loading text while dispense notification is being submitted 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Loading...</label></h1><label class=\\"nhsuk-label\\">Sending dispense notification.</label>"`;

exports[`Displays loading text while prescription data is being requested 1`] = `"<h1 class=\\"nhsuk-label-wrapper\\"><label class=\\"nhsuk-label nhsuk-label--xl\\">Loading...</label></h1><label class=\\"nhsuk-label\\">Retrieving prescription details.</label>"`;
